File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String,Integer> people = new HashMap<String,Integer>();
        
        for(String p : participant){
            if(people.containsKey(p)){
                people.put(p,(people.get(p))+1);
            } else {
                people.put(p,1);
            }
        }
        
        for(String c : completion){
            if(people.containsKey(c)){
                people.put(c,(people.get(c))-1);
            }
        }
        
        Iterator<String> iterator = people.keySet().iterator();
        while(iterator.hasNext()){
            String key = iterator.next();
            if(people.get(key)!=0) answer = key;
        }
        
        return answer;
    }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = '';
    const people = new Map();
    
    participant.forEach((name) => people.set(name, (people.get(name) || 0) + 1));
    completion.forEach((name) => people.set(name, people.get(name) -1));
    
    for(let name of people.keys()){
        if(people.get(name) != 0){
            answer = name;
        }        
    }    
    return answer;
}